services/ext/api.go

```diff
@@ -408,8 +402,8 @@ func (api *PublicAPI) SerializedCommunities(parent context.Context) ([]json.RawM
 }
 
 // JoinedCommunities returns a list of communities that the user has joined
-func (api *PublicAPI) JoinedCommunities(parent context.Context) ([]*communities.Community, error) {
-	return api.service.messenger.JoinedCommunities()
+func (api *PublicAPI) JoinedCommunities(parent context.Context) ([]json.RawMessage, error) {
```

Comment:
I don't like the concept of "serialized" communities as it makes the return type less self-explanatory. I would rely on `MarshalJSON` inteface as before. For that, we would need to inject media server interface into community object, but that's much better imho.

Reply:
I see 2 options here:
- First, inject httpServer into the community. But then whenever you get a community and want to pass it via api, you have to double check that c.httpServer is set
```
type Community struct {
    // ...
    httpServer *server.MediaServer
}

func (c *Community) SetMediaServer(server *server.MediaServer) {
    c.httpServer = server
}

// example:
func (m *Messenger) Communities() ([]*communities.Community, error) {
    communities, err := m.communitiesManager.All()
    if err != nil {
        return nil, err
    }
    for _, community := range communities {
        community.SetMediaServer(m.httpServer)
    }
    return communities, nil
}  

```
- Second, introduce the SerializableCommunity (updated PR).

Reply:
I am still in favor of first option :sweat_smile:  `SerializedCommunities` even in wrapper looks to me very strange and artificial.

> But then whenever you get a community and want to pass it via api, you have to double check that c.httpServer is set

If it is ensured at the time of community object creation, then we don't need to check it on every serialization, something like:
```diff
-func New(config Config, timesource common.TimeSource, encryptor DescriptionEncryptor) (*Community, error) {
+func New(config Config, timesource common.TimeSource, encryptor DescriptionEncryptor, mediaServer *server.MediaServer) (*Community, error) {
     if config.MemberIdentity == nil {
         return nil, errors.New("no member identity")
     }
@@ -80,6 +80,10 @@ func New(config Config, timesource common.TimeSource, encryptor DescriptionEncry
         return nil, errors.New("no timesource")
     }
 
+    if mediaServer == nil {
+        return nil, errors.New("no mediaserver")
+    }
+
     if config.Logger == nil {
         logger, err := zap.NewDevelopment()
         if err != nil {
@@ -92,7 +96,7 @@ func New(config Config, timesource common.TimeSource, encryptor DescriptionEncry
         config.CommunityDescription = &protobuf.CommunityDescription{}
     }
 
-    return &Community{config: &config, timesource: timesource, encryptor: encryptor}, nil
+    return &Community{config: &config, timesource: timesource, encryptor: encryptor, mediaServer: mediaServer}, nil
 }
```

Note: we should use interface rather than exact type here.

Reply:
I see, it makes sense to have everything needed to display the object. Fixed

In my opinion, the downside of this incremental approach is that the community currently feels cohesive and intertwined with a lot of things. I'd like to see its logic broken down into simpler components (chat, members, permissions, serialisation, ...).